src/viam/sdk/robot/client.hpp

```diff
@@ -81,54 +132,50 @@ class RobotClient {
     ///
     /// Because the return type here is a `Resource`, the user will need to manually
     /// cast to the desired type.
-    std::shared_ptr<Resource> resource_by_name(const ResourceName& name);
+    std::shared_ptr<Resource> resource_by_name(const Name& name);
 
     template <typename T>
     /// @brief Lookup and return a `shared_ptr` to a resource of the requested type.
     /// @param name The ordinary name of the resource.
     /// @throws `std::runtime_error` if the requested resource doesn't exist or is the wrong type.
     /// @return a `shared_ptr` to the requested resource.
     std::shared_ptr<T> resource_by_name(std::string name) {
-        ResourceName r;
         API api = API::get<T>();
-        *r.mutable_namespace_() = api.type_namespace();
-        *r.mutable_type() = api.resource_type();
-        *r.mutable_subtype() = api.resource_subtype();
-        *r.mutable_name() = std::move(name);
+        Name r = Name::from_string(api.to_string() + "/" + name);
```

Comment:
This is more "assembling a string to have `Name::from_string` decompose it". Is there a better `Name` API to use here?